### DISCUSSION TITLE
Link to section in notebook

### DISCUSSION THREAD

**Discussion by icweaver** [fonsp/Pluto.jl#1844] [Questions about using Pluto] [answered]
Is there a way to do this for hosted notebooks? For example, I would like to link to the section header "3. Generate the report ✏️" in [this notebook](https://icweaver.github.io/WASP-50b/html/comp_stars.jl.html). Trying something like:

```
https://icweaver.github.io/WASP-50b/html/comp_stars.jl.html#generate-the-report
```
or
```
https://icweaver.github.io/WASP-50b/html/comp_stars.jl.html#1f2afbc8-5e04-4f18-b047-90775443d572
```

doesn't seem to work for me, but I guess that is because there is no div id here, except for in `pluto-cell`?

![Screenshot from 2022-01-22 11-11-01](https://user-images.githubusercontent.com/25312320/150646527-5dc5b3b8-d2cd-4db3-b776-ea7a7b4ba9e0.png)

No idea how that works though!

**Comment by ctrekker** [comment, marked as answer]
I don't think this is possible without some sort of special implementation on Pluto's side. The `#1f2afbc8-5e04-4f18-b047-90775443d572` in the URL tells the browser to look for an element with `id=1f2afbc8-5e04-4f18-b047-90775443d572` somewhere in the DOM tree, but this element doesn't actually exist when the page is first fetched, which can be seen with Chrome's view source function. Instead Pluto generates all the DOM you see in inspect element right _after_ the page loads.

A very dirty way to fix this would be to place this code snippet at the end of the HTML `<head>` block. A much cleaner solution could be implemented within Pluto itself by doing the same thing that's done in the snippet, but right after the Preact render call.

```html
<script>
        window.addEventListener('load', () => {
            setTimeout(() => {
                if(window.location.hash) {
                    const scrollToId = window.location.hash.slice(1);
                    const scrollToEl = document.getElementById(scrollToId);
                    if(scrollToEl) {
                        scrollToEl.scrollIntoView({block: 'start'});
                    }
                }
            }, 100);
        });
</script>
```

All this does is wait 100 milliseconds so that hopefully the page is loaded by then, and then searches the DOM for an element which has the ID provided in the location hash (which is what the `#` denotes). If this element exists, then the JS will scroll to it.

**Reply by icweaver** [reply]
Ah, I see. Thanks for such a clear explanation and potential workarounds @ctrekker!

**Comment by fonsp** [comment]
This is actually a feature of https://github.com/JuliaPluto/MarkdownLiteral.jl! It automatically generates IDs for all headers, and you can link to IDs with the `#`.

E.g.
```julia
# ╔═╡ 2cb491d7-6b33-424c-83de-cce087a904a8
import MarkdownLiteral: @mdx

# ╔═╡ 2db953ae-b4d9-49cf-8b24-58425a71f5b7
@mdx "Take a look at [the next section](#setting-up-the-game-pieces)"

# ╔═╡ 95fbd0d2-a2b9-11ea-0682-fdf783251797
@mdx """
## Setting up the game pieces

What does a Julia implementation look like? We're not really interested in writing code that will manipulate physical disks. lkajsdlkjsdlkfj
"""
```


https://user-images.githubusercontent.com/6933510/151026907-bdeb0911-b3b5-42d8-9fa9-4d38668be559.mov

**Reply by fonsp** [reply]
But it would be nice to have an easy way to know the ID without guessing or opening the web dev tools!

**Reply by icweaver** [reply]
Oh what the heck, what can't MarkdownLiteral.jl do?

**Reply by MichaelHatherly** [reply]
Yeah, the slugification is defined in https://github.com/MichaelHatherly/CommonMark.jl/blob/efcbbc8bd1f19c96fce7fa7bcf06f6c646dd8567/src/extensions/autoidentifiers.jl#L23-L30, but it would be nice to surface that in the actual rendered headers in some way, perhaps like is done in github markdown with the anchor on hover thing. One difficulty though is that the parser only knows about the headers in the chunk of text that it parses, rather than all the other cells that contain markdown, which means that making all header IDs unique isn't really possible. (A single chunk of rendered markdown will add auto incremented numbers to headers with the same slugs.)
